server/core/src/main/java/dev/slimevr/protocol/ProtocolHandler.kt

```diff
@@ -0,0 +1,25 @@
+package dev.slimevr.protocol
+
+import java.util.function.BiConsumer
+
+abstract class ProtocolHandler<H> {
+	@JvmField
+	val handlers: Array<BiConsumer<GenericConnection, H>?>
+
+	init {
+		this.handlers = arrayOfNulls(this.messagesCount())
+	}
+
+	abstract fun onMessage(conn: GenericConnection, message: H)
+
+	abstract fun messagesCount(): Int
+
+	fun registerPacketListener(packetType: Byte, consumer: BiConsumer<GenericConnection, H>?) {
+		if (handlers[packetType.toInt()] != null) {
+			handlers[packetType.toInt()] =
+				consumer?.let { handlers[packetType.toInt()]!!.andThen(it) }
+		} else {
+			handlers[packetType.toInt()] = consumer
+		}
+	}
```

Comment:
If consumer is null then it'll stop the existing handler from functioning. Why not just make it not nullable?
```suggestion
	fun registerPacketListener(packetType: Byte, consumer: BiConsumer<GenericConnection, H>) {
		val packetInt = packetType.toInt()
		if (handlers[packetInt] != null) {
			handlers[packetInt] = handlers[packetInt]!!.andThen(consumer)
		} else {
			handlers[packetInt] = consumer
		}
	}
```